pkgs/applications/graphics/meshlab-unstable/default.nix

```diff
@@ -0,0 +1,127 @@
+{
+  mkDerivation,
+  lib,
+  fetchFromGitHub,
+  libGLU,
+  qtbase,
+  qtscript,
+  qtxmlpatterns,
+  lib3ds,
+  bzip2,
+  muparser,
+  eigen,
+  glew,
+  gmp,
+  levmar,
+  qhull,
+  cmake,
+  cgal,
+  boost,
+  mpfr,
+  xercesc,
+  tbb,
+  embree,
+  libigl,
+  corto,
+  openctm,
+  structuresynth,
+}:
+
+let
+  tinygltf-src = fetchFromGitHub {
+    owner = "syoyo";
+    repo = "tinygltf";
+    rev = "v2.6.3";
+    hash = "sha256-IyezvHzgLRyc3z8HdNsQMqDEhP+Ytw0stFNak3C8lTo=";
+  };
+  # requires submodules to build
+  lib3mf-src = fetchFromGitHub {
+    owner = "3MFConsortium";
+    repo = "lib3mf";
+    rev = "v2.3.2";
+    fetchSubmodules = true;
+    hash = "sha256-pKjnN9H6/A2zPvzpFed65J+mnNwG/dkSE2/pW7IlN58=";
+  };
+in
+mkDerivation {
+  pname = "meshlab-unstable";
+  version = "2023.12-unstable-2025-02-21";
+
+  src = fetchFromGitHub {
+    owner = "cnr-isti-vclab";
+    repo = "meshlab";
+    # note that this is in branch devel
+    rev = "72142583980b6dbfc5b85c6cca226a72f48497a9";
+    sha256 = "1q7qga4d82pvpcbsp9pi2i7nzdbflhp6q0d3y31kpch9r3r9pzks";
+    # needed for an updated version of vcg in their submodule
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [
+    libGLU
+    qtbase
+    qtscript
+    qtxmlpatterns
+    lib3ds
+    bzip2
+    muparser
+    eigen
+    glew
+    gmp
+    levmar
+    qhull
+    cgal
+    boost
+    mpfr
+    xercesc
+    tbb
+    embree
+    libigl
+    corto
+    openctm
+    structuresynth
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
+  preConfigure = ''
+    mkdir src/external/downloads
+    cp -r --no-preserve=mode,ownership ${lib3mf-src} src/external/downloads/lib3mf-2.3.2
+
+    substituteAll ${../meshlab/meshlab.desktop} resources/linux/meshlab.desktop
```

Comment:
Instead of `../`, which will fail a future by-name migration, please either 1) copy the file here, 2) use `makeDesktopItem`, or 3) expose the original file in `passthru` of `meshlab` and refer to it here

Reply:
Fixed